core/src/main/java/com/datastax/oss/driver/internal/core/data/DefaultTupleValue.java

```diff
@@ -93,6 +95,44 @@ private void readObject(ObjectInputStream stream) throws InvalidObjectException
     throw new InvalidObjectException("Proxy required");
   }
 
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+
+    if (o == null || !(o instanceof TupleValue)) return false;
+    DefaultTupleValue that = (DefaultTupleValue) o;
+
+    if (this.protocolVersion() != that.protocolVersion()) return false;
```

Comment:
In practice all instances of `ProtocolVersion` are enum values so this works, but since in theory there could be other implementations, this should use `equals()`.

Unrelated: I've noticed that `hashCode` doesn't use the protocol version, but that's not necessarily an issue: it doesn't break the rule that equal objects must have the same hash code, and in practice different protocol versions will be rare, so I'm ok with it.